packages/block-library/src/avatar/index.php

```diff
@@ -16,67 +16,14 @@
 function render_block_core_avatar( $attributes, $content, $block ) {
 	$size               = isset( $attributes['size'] ) ? $attributes['size'] : 96;
 	$wrapper_attributes = get_block_wrapper_attributes();
+	$border_attributes  = get_block_core_avatar_border_attributes( $attributes );
 
-	$image_styles = array();
-
-	// Add border width styles.
-	$has_border_width = ! empty( $attributes['style']['border']['width'] );
-
-	if ( $has_border_width ) {
-		$border_width   = $attributes['style']['border']['width'];
-		$image_styles[] = sprintf( 'border-width: %s;', esc_attr( $border_width ) );
-	}
-
-	// Add border radius styles.
-	$has_border_radius = ! empty( $attributes['style']['border']['radius'] );
-
-	if ( $has_border_radius ) {
-		$border_radius = $attributes['style']['border']['radius'];
-
-		if ( is_array( $border_radius ) ) {
-			// Apply styles for individual corner border radii.
-			foreach ( $border_radius as $key => $value ) {
-				if ( null !== $value ) {
-					$name = _wp_to_kebab_case( $key );
-					// Add shared styles for individual border radii.
-					$border_style   = sprintf(
-						'border-%s-radius: %s;',
-						esc_attr( $name ),
-						esc_attr( $value )
-					);
-					$image_styles[] = $border_style;
-				}
-			}
-		} else {
-			$border_style   = sprintf( 'border-radius: %s;', esc_attr( $border_radius ) );
-			$image_styles[] = $border_style;
-		}
-	}
-
-	// Add border color styles.
-	$has_border_color = ! empty( $attributes['style']['border']['color'] );
-
-	if ( $has_border_color ) {
-		$border_color   = $attributes['style']['border']['color'];
-		$image_styles[] = sprintf( 'border-color: %s;', esc_attr( $border_color ) );
-	}
-
-	// Add border style (solid, dashed, dotted ).
-	$has_border_style = ! empty( $attributes['style']['border']['style'] );
-
-	if ( $has_border_style ) {
-		$border_style   = $attributes['style']['border']['style'];
-		$image_styles[] = sprintf( 'border-style: %s;', esc_attr( $border_style ) );
-	}
-
-	// Add border classes to the avatar image for both custom colors and palette colors.
-	$image_classes = '';
-	if ( $has_border_color || isset( $attributes['borderColor'] ) ) {
-		$image_classes .= 'has-border-color';
-	}
-	if ( isset( $attributes['borderColor'] ) ) {
-		$image_classes .= ' has-' . $attributes['borderColor'] . '-border-color';
-	}
+	$image_classes = ! empty( $border_attributes['class'] )
+		? "wp-block-avatar__image {$border_attributes['class']}"
+		: 'wp-block-avatar__image';
+	$image_styles  = ! empty( $border_attributes['style'] )
+		? sprintf( ' style="%s"', safecss_filter_attr( $border_attributes['style'] ) )
```

Comment:
The style engine already filters attributes, so I don't think we need the `safecss_filter_attr` call here?

To dig in, `wp_style_engine_get_styles` calls compile_css [here](https://github.com/WordPress/gutenberg/blob/8c79d908ac4ea5775a9f3c34e67442215a69b630/packages/style-engine/style-engine.php#L57), which [then calls](https://github.com/WordPress/gutenberg/blob/758372afab62b8665277776d860f5acf5da3c41b/packages/style-engine/class-wp-style-engine.php#L538) [`get_declarations_string`](https://github.com/WordPress/gutenberg/blob/388e5d23e327e13b7e88f60268b99d89e761bf79/packages/style-engine/class-wp-style-engine-css-declarations.php#L174) which calls [`filter_declaration`](https://github.com/WordPress/gutenberg/blob/388e5d23e327e13b7e88f60268b99d89e761bf79/packages/style-engine/class-wp-style-engine-css-declarations.php#L141), so it should already be filtered.

Reply:
Thanks for flagging this. 

On closer inspection the `get_avatar` function doesn't escape the attributes passed via the `extra_attr` arg so I've replaced the superfluous call to `safecss_filter_attr` with one to `esc_attr`. 

In addition, a couple of inline comments have been added explaining the escaping. Hopefully that is sufficient for anyone else in future adding something here to know how and what they'll need to escape.

Reply:
Thanks for the update! Looks good 👍